src/HashMap.mo

```diff
@@ -219,4 +230,18 @@ module {
     h2
   };
 
+  /// The mutable state of a `HashMap`
+  public type S<K,V> = {
```

Comment:
Perhaps clearer to move this to the top, since `S` is already used in the HashMap class (I was confused at first where `S` is coming from).

Reply:
Actually, it could be a `public type` of the class itself, right? Then it could be near the end, but part of the class (if optimizing for documentation reader's preferred order, not implementation reader's.)

Reply:
> Actually, it could be a public type of the class itself, right?

My understanding is that the type system (still) does not permit type members of classes, only modules, or maybe that changed?

Reply:
Oh, not sure. Why should it not? But if not, ignore me.